### PR TITLE
indicator-application-gtk3: 12.10.0 -> 12.10.1

### PR DESCRIPTION
**Pull request by worldofpeace**
###### Motivation for this change
I guess elementary needs this version for the ayatana indicators.
I now includes a systemd user service and xdg autostart.

One thing I noticed is that it wanted to install its libraries to /nix/store/*-libindicator-gtk3-*/lib/indicators3/7. 
I hope that's not important.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
471815040
471811400
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---
cc @jtojnar

### CODE REVIEW

**Comment by srhb** [comment]
Retriggering eval due to master failure
@GrahamcOfBorg eval

**Comment by jtojnar** [comment]
On a side note, there is a new fork of the indicators spearheaded by the Debian’s Ayatana team. We might want to package it too and ask elementary for its support.

* https://alioth-lists.debian.net/pipermail/pkg-ayatana-devel/2018-March/001769.html
* https://wiki.debian.org/Ayatana/IndicatorsTransition
* https://github.com/AyatanaIndicators
* https://debconf17.debconf.org/talks/168/
* https://gobby.debian.org/export/debconf17/talk/ayatana-indicators

File: pkgs/development/libraries/indicator-application/gtk3.nix [hunk 1]
```diff
@@ -1,39 +1,36 @@
...
+  name = "${pname}-gtk3-${version}";
+
+  src = fetchbzr {
+    url = "https://code.launchpad.net/~indicator-applet-developers/${pname}/trunk.17.04";
+    rev = "260";
+    sha256 = "1f0jdyqqb5g86zdpbcyn16x94yjigsfiv2kf73dvni5rp1vafbq1";
   };
 
-  nativeBuildInputs = [ pkgconfig autoconf ];
+  nativeBuildInputs = [ pkgconfig autoreconfHook systemd ];
```
Comment (timokau) [inline pkgs/development/libraries/indicator-application/gtk3.nix]: Is systemd actually a `nativeBuildInput`? If it is used at runtime, it should be a `buildInput`.

**Comment by worldofpeace** [comment]
@jtojnar I think that's a good idea and something they would be interested in :+1:

File: pkgs/development/libraries/indicator-application/gtk3.nix [hunk 2]
```diff
@@ -1,39 +1,36 @@
...
-    done
-    substituteInPlace src/Makefile.in \
+    substituteInPlace data/Makefile.am \
+      --replace "/etc/xdg/autostart" "$out/etc/xdg/autostart"
+
+    substituteInPlace src/Makefile.am \
       --replace 'applicationlibdir = $(INDICATORDIR)' "applicationlibdir = $out/lib"
+
+    substituteInPlace configure.ac \
+      --replace 'SYSTEMD_USERDIR=`$PKG_CONFIG --variable=systemduserunitdir systemd`' 'SYSTEMD_USERDIR="$out/lib/systemd/user"'
```
Comment (jtojnar) [inline pkgs/development/libraries/indicator-application/gtk3.nix]: Can you try setting the `PKG_CONFIG_SYSTEMD_SYSTEMDUSERUNITDIR = "$(out)/lib/systemd/user";` attribute in the expression?

File: pkgs/development/libraries/indicator-application/gtk3.nix [hunk 3]
```diff
@@ -1,39 +1,36 @@
...
-    for f in {configure,ltmain.sh,m4/libtool.m4}; do
-      substituteInPlace $f \
-        --replace /usr/bin/file ${file}/bin/file
-    done
-    substituteInPlace src/Makefile.in \
+    substituteInPlace data/Makefile.am \
+      --replace "/etc/xdg/autostart" "$out/etc/xdg/autostart"
+
+    substituteInPlace src/Makefile.am \
       --replace 'applicationlibdir = $(INDICATORDIR)' "applicationlibdir = $out/lib"
```
Comment (jtojnar) [inline pkgs/development/libraries/indicator-application/gtk3.nix]: Same here with `PKG_CONFIG_AYATANA_INDICATOR3_0_4_INDICATORDIR`

Comment (worldofpeace) [reply pkgs/development/libraries/indicator-application/gtk3.nix]: Lol I tried that early on but I struggle to get the name right even when I look at the pc file :)

File: pkgs/development/libraries/indicator-application/gtk3.nix [hunk 4]
```diff
@@ -1,39 +1,30 @@
...
+  name = "${pname}-gtk3-${version}";
+
+  src = fetchbzr {
+    url = "https://code.launchpad.net/~indicator-applet-developers/${pname}/trunk.17.04";
+    rev = "260";
+    sha256 = "1f0jdyqqb5g86zdpbcyn16x94yjigsfiv2kf73dvni5rp1vafbq1";
   };
 
-  nativeBuildInputs = [ pkgconfig autoconf ];
+  nativeBuildInputs = [ pkgconfig systemd autoreconfHook ];
```
Comment (jtojnar) [inline pkgs/development/libraries/indicator-application/gtk3.nix]: Still in native.

Comment (worldofpeace) [reply pkgs/development/libraries/indicator-application/gtk3.nix]: Without it there was no $out/lib/systemd

Comment (jtojnar) [reply pkgs/development/libraries/indicator-application/gtk3.nix]: But should not it be in `buildInputs` instead of `nativeBuildInputs`?

Comment (worldofpeace) [reply pkgs/development/libraries/indicator-application/gtk3.nix]: Oh right. Done.

**Comment by timokau** [comment]
@GrahamcOfBorg build indicator-application-gtk3

**Comment by jtojnar** [comment]
Just looking into the lib location.

**Comment by jtojnar** [comment]
I do not see anything out of ordinary. Could you clarify what you mean by

> One thing I noticed is that it wanted to install its libraries to /nix/store/-libindicator-gtk3-/lib/indicators3/7.

**Comment by worldofpeace** [comment]
That's because it wanted to install there using the libindicators `indicatordir`.
It shouldn't be a problem and I've done it virtually as before 

 * https://github.com/NixOS/nixpkgs/pull/45052/files#diff-755e3688d8e01d3e67fb763c8315c8aeL36
* https://github.com/NixOS/nixpkgs/pull/45052/files#diff-755e3688d8e01d3e67fb763c8315c8aeR42

**Comment by jtojnar** [comment]
Yes, that should probably be fixed to `$(out)/lib/indicators3/7/` since this is not a library but more of a plug-in.

**Review by timokau — APPROVED** [review]

File: pkgs/development/libraries/indicator-application/gtk3.nix [hunk 5]
```diff
@@ -47,7 +38,16 @@ stdenv.mkDerivation rec {
     "localstatedir=\${TMPDIR}"
   ];
 
-  meta = {
+  PKG_CONFIG_SYSTEMD_SYSTEMDUSERUNITDIR = "$(out)/lib/systemd/user";
+  PKG_CONFIG_INDICATOR3_0_4_INDICATORDIR = "$(out)/lib";
```
Comment (jtojnar) [inline pkgs/development/libraries/indicator-application/gtk3.nix]: Needs to be changed to `$(out)/lib/indicators3/7/` as per `indicator3-0.4.pc` file.

**Review by jtojnar — CHANGES_REQUESTED** [review]

**Comment by worldofpeace** [comment]
Ok, great :)

**Pull request merged**